### PR TITLE
Implemented getBrowserInfo (closes #42)

### LINKED ISSUE #42: Implement getBrowserInfo api function
`getBrowserInfo(browser)`

where `browser` is a string:

- if it's an `browser alias` (like `chrome` or `ie`) just return the information from `getInstallations` for this alias;
- if it's a file path check file existing. Then check the path with regexp to find some browser alias in there. If found, get the `browserInfo` structure (cmd, templates) from `getInstallations` by the alias and return it with current path. If not found return default browser info (for example with something like `start filePath` for windows etc.)

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin 
Also @VasilyStrelyaev, please check new error message in `src/messages.js`, new test name in `test/tests/get-browser-info-test.js` and check/write new documentation for `src/api/get-browser-info`.

### CODE REVIEW

File: src/api/get-browser-info.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    if (installations[browserAsAlias])
+        return installations[browserAsAlias];
+
+    var fileExists = await exists(browser);
+
+    if (!fileExists)
+        throw new Error(getText(MESSAGES.unableToFindBrowser, browser));
+
+    var detectedAliases = Object.keys(ALIASES).filter(alias => ALIASES[alias].nameRe.test(browser));
```
Comment (inikulin) [inline src/api/get-browser-info.js]: Use https://www.npmjs.com/package/array-find util here

File: src/api/get-browser-info.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    var browserAsAlias = browser.trim().toLowerCase();
+
+    if (installations[browserAsAlias])
+        return installations[browserAsAlias];
+
+    var fileExists = await exists(browser);
+
+    if (!fileExists)
+        throw new Error(getText(MESSAGES.unableToFindBrowser, browser));
```
Comment (inikulin) [inline src/api/get-browser-info.js]: It more like `MESSAGES.browserPathDoesntExist` or something like that. \cc @VasilyStrelyaev

File: src/messages.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 export const MESSAGES = {
     browserPathNotSet:       'Unable to run the browser. The browser path or command template is not specified.',
     unableToRunBrowser:      'Unable to run the browser. The file at {path} does not exist or is not executable.',
+    unableToFindBrowser:     'Unable to find the browser. "{path}" is not a browser alias or a path to browser executable.',
```
Comment (inikulin) [inline src/messages.js]: Just stylistic remark: `"{browser}" "{path}" is not a browser alias or a path to browser executable.`

File: src/api/get-browser-info.js (hunk 2 above)
Comment (inikulin) [inline src/api/get-browser-info.js]: **deleted comment**

File: test/tests/get-browser-info-test.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('getBrowserInfo', function () {
+    it('Should raise an error if browser is not an alias or a path', function (done) {
+        browserNatives
+            .getBrowserInfo('ffuuuuu')
+            .then(function () {
+                throw new Error('Promise rejection expected');
+            })
+            .catch(function (err) {
+                expect(err.message).eql(messages.getText(messages.MESSAGES.unableToFindBrowser, 'ffuuuuu'));
```
Comment (inikulin) [inline test/tests/get-browser-info-test.js]: Use string representation of the resulting messsage in tests.

**Comment by inikulin** [comment]
\r-

**Comment by AndreyBelym** [comment]
FPR

File: test/tests/get-browser-info-test.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('getBrowserInfo', function () {
+    it('Should raise an error if browser is not an alias or a path', function (done) {
+        browserNatives
+            .getBrowserInfo('ffuuuuu')
+            .then(function () {
+                throw new Error('Promise rejection expected');
+            })
+            .catch(function (err) {
+                expect(err.message).eql('Unable to find the browser. ' + '' +
```
Comment (inikulin) [inline test/tests/get-browser-info-test.js]: Strange concatenation. This string is not that long, you can keep it whole on the single line

File: test/tests/open-test.js [hunk 6]
```diff
@@ -14,7 +13,8 @@ describe('open', function () {
                 throw new Error('Promise rejection expected');
             })
             .catch(function (err) {
-                expect(err.message).eql(messages.getText(messages.MESSAGES.browserPathNotSet));
+                expect(err.message).eql('Unable to run the browser. ' +
```
Comment (inikulin) [inline test/tests/open-test.js]: Same here, keep it as is

File: test/tests/open-test.js [hunk 7]
```diff
@@ -35,7 +35,8 @@ describe('open', function () {
                 throw new Error('Promise rejection expected');
             })
             .catch(function (err) {
-                expect(err.message).eql(messages.getText(messages.MESSAGES.unableToRunBrowser, browserInfo.path));
+                expect(err.message).eql('Unable to run the browser. ' +
```
Comment (inikulin) [inline test/tests/open-test.js]: Here you need to split lines, but try to achieve equal lines length for better readability.

**Comment by inikulin** [comment]
\r-

**Comment by AndreyBelym** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

File: src/messages.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 export const MESSAGES = {
     browserPathNotSet:       'Unable to run the browser. The browser path or command template is not specified.',
     unableToRunBrowser:      'Unable to run the browser. The file at {path} does not exist or is not executable.',
+    unableToFindBrowser:     'Unable to find the browser. "{browser}" is not a browser alias or a path to browser executable.',
```
Comment (VasilyStrelyaev) [inline src/messages.js]: Unable to find the browser. "{browser}" is not a browser alias or path to an executable file.

File: test/tests/get-browser-info-test.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+var expect         = require('chai').expect;
+var browserNatives = require('../../lib/index');
+
+describe('getBrowserInfo', function () {
+    it('Should raise an error if browser is not an alias or a path', function (done) {
```
Comment (VasilyStrelyaev) [inline test/tests/get-browser-info-test.js]: Should raise an error if the browser ID is not an alias or file path

File: src/api/get-browser-info.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+//API
+/**
+ * Returns information about the browser specified by browser alias ('chrome', 'ff')
+ * or a path to browser executable file.
+ * @function
+ * @async
+ * @name getBrowserInfo
+ * @param {string} browser - Browser alias or a path to browser executable file.
+ * @returns {BrowserInfo} Object containing information about the browsers installed on the machine.
+ */
```
Comment (VasilyStrelyaev) [inline src/api/get-browser-info.js]:
```
//API
+/**
+ * Returns information about the specified browser. 
+ * @function
+ * @async
+ * @name getBrowserInfo
+ * @param {string} browser - A browser alias ('chrome', 'ff', etc.) or a path to the browser's executable file.
+ * @returns {BrowserInfo} An object that contains information about the specified browser.
+ */
```

**Comment by AndreyBelym** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**